src/client/api/container.c

```diff
@@ -508,28 +508,61 @@ daos_cont_list_attr(daos_handle_t coh, char *buf, size_t *size,
 	return dc_task_schedule(task, true);
 }
 
+static int
+free_name(tse_task_t *task, void *args)
+{
+	char *name = *(char **)args;
+	free(name);
+	return 0;
+}
+
 int
 daos_cont_get_attr(daos_handle_t coh, int n, char const *const names[],
 		   void *const values[], size_t sizes[], daos_event_t *ev)
 {
 	daos_cont_get_attr_t	*args;
 	tse_task_t		*task;
-	int			 rc;
+	int			 i, rc;
+	char			**new_names;
 
 	DAOS_API_ARG_ASSERT(*args, CONT_GET_ATTR);
 
+	D_ALLOC_ARRAY(new_names, n);
+	if (!new_names)
+		return -DER_NOMEM;
+
 	rc = dc_task_create(dc_cont_get_attr, NULL, ev, &task);
 	if (rc)
 		return rc;
 
+	for (i = 0 ; i < n ; i++) {
+		/* no easy way to determine if a name storage address is likely
+		 * to cause an EFAULT during memory registration, so duplicate
+		 * name in heap
+		 */
+		new_names[i] = strdup(names[i]);
```

Comment:
this will likely trigger an SDL check i think.
name is an input from external users and it needs to be bounded. should use strndup with DAOS_ATTR_NAME_MAX if one exists.
this is probably not your fault here if the max name value does not exist. we should have added it before.

Reply:
yea i mentioned this is probably an oversight with whomever did this API.
you can handle this in your PR or we can create a ticket.
if you want to add it here, i would suggest same max len as NAME_MAX (255)